### DISCUSSION TITLE
`ColumnEnsembleForecaster` does not keep column names in predictions

### DISCUSSION THREAD

**Discussion by Yassine97HF** [alan-turing-institute/sktime#3423] [Q&A]
Hi everyone  !

i want to share with you my problem,
 i want to fit this pipeline:

```python 
from sktime.datasets import load_longley
from sktime.forecasting.compose import ColumnEnsembleForecaster
from sktime.forecasting.fbprophet import Prophet
from sktime.forecasting.theta import ThetaForecaster
from sktime.forecasting.base import ForecastingHorizon
from sktime.forecasting.compose import ForecastX 

y, X = load_longley()
y= y.reset_index(drop=True)
X = X.reset_index(drop=True)
fh = ForecastingHorizon([1, 2, 3])
pipeline = ForecastX(
                      forecaster_X=ColumnEnsembleForecaster(ThetaForecaster()),
                      forecaster_y=Prophet()
                    )
pipeline=pipeline.fit(y,X,fh)
pipeline.predict(fh=fh)
```

When i try to make a predictions i got this error:

![image](https://user-images.githubusercontent.com/102746985/189979820-0f4b1186-d8f6-4d4e-8783-45ae7abf86e4.png)
 
So i tried the following code to have an idea how `ColumnEnsembleForecaster` run:

```python
 forecaster = ColumnEnsembleForecaster(ThetaForecaster())
forecaster=forecaster.fit(y=X)
forecaster.predict(fh = [1, 2, 3])
```

then i got a dataframe without names of columns :

![image](https://user-images.githubusercontent.com/102746985/189980877-a9e2476c-f2e1-4938-8231-9db960dfb1f9.png)

MY QUESTION:

there is a way to force ` ColumnEnsembleForecaster` to keep column names?

**Comment by fkiraly** [comment]
There was a bug a few versions ago that lost the column names in case of vectorization over columns.

You must be having an old `sktime` version since the code above runs without problems for me. Current version is 0.13.2.
I suspect you installed via conda which is outdated, see https://github.com/alan-turing-institute/sktime/issues/2921 for the problems we have with conda currently. `pip install sktime` should help you.

FYI: in the most up-to-date version you no longer need to wrap in the `ColumnEnsembleForecaster` to make the estimator multivariate, it does that automatically.
